### ISSUE TITLE
Hints doesn't work on Archlinux Security page.

### ISSUE THREAD

**Issue by 0a-a** [fanglingsu/vimb#531]
### Steps to reproduce

1. Open the site [https://security.archlinux.org/](https://security.archlinux.org/)
2. Trigger any hints (`f`, `;x`, `;o`, etc)

### Expected behaviour

Elements get highlighted and the keyhints show up
Highlight work, but no keyhints show up.
### Actual behaviour
![](https://ptpb.pw/PzaN.png)
```
$ vimb --bug-info
Commit:          3.3.0-3-g502b49a
WebKit compile:  2.22.2
WebKit run:      2.22.2
GTK compile:     3.24.1
GTK run:         3.24.1
libsoup compile: 2.64.2
libsoup run:     2.64.2
Extension dir:   /usr/lib/vimb
```
Was the same with vimb version 3.2.

**Comment by fanglingsu** [comment]
@0a-a Thank you for reporting this issue. I can reproduce this. The hint-labels are kept as `display: none`, the default value when they are created. I assume that this is related to the error in the webinspector "Refused to apply a stylesheet because its hash, its nonce, or 'unsafe-inline' does not appear in the style-src directive of the Content Security Policy.". 
So that the Stylechanges of vimb could not be applied to the hinted elements or hint labels because the inline styles are not evaluated. We could remove the `disaply:none` part, but the inline set positions for the labels would also by denied. I'm note sure if there is a a working solution for this issue. Maybe we could do some shadow-DOM stuff to create the hintlabels.

**Comment by fanglingsu** [comment]
Yes, this seems to be the same issue with the Content-Security-Policy.

**Comment by fanglingsu** [comment]
I don't think we can do anything to bypass this securoty mechanism. OK we could, by removing the CSP headers before they are processed by webkit. But I don't want to remove this extra pice of security.
I've checked if it would be possible to style shadow DOM nodes, but this is also not allowed.
